### PR TITLE
Release 1.1.2 #443 

### LINKED ISSUE #443: new release 1.1.2
Release 1.1.2 with fixed model and retrained pattern 'var in the middle'

### PR DESCRIPTION
**Pull request by lyriccoder**
fix #443

### CODE REVIEW

**Comment by lyriccoder** [comment]
Here is the output:
```
D:\git\aibolit\venv\Scripts\python.exe D:/git/aibolit/aibolit/__main__.py recommend --full --folder D:\target\0001\fast
Version 1.1.1 is available, but you are using 0.0.0
Show all patterns
D:\target\0001\fast\Configuration.java score: 13.200000000000001
D:\target\0001\fast\Configuration.java[294]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[391]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[393]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[694]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[729]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[749]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[751]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[764]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[766]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[791]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[794]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[809]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[1129]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[1150]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[1156]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[1172]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[1175]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[1299]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[1334]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[1379]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[1382]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[1391]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[1396]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[1424]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[1437]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[1457]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[1463]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[1505]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[1508]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[1558]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[1561]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[1583]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[1630]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[1659]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[2021]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[2042]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[2050]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[2050]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[2054]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[2092]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[2103]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[2105]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[2182]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[2207]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[2272]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[2354]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[2373]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[2379]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[2381]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[2389]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[2389]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[2392]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[2416]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[2420]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[2450]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[2481]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[2534]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[2534]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[2569]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[2588]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[2597]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[2601]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[2661]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[2691]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[2693]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[2822]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[2846]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[2872]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[2874]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[2878]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[2880]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[2884]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[2961]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[2979]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[3002]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[3024]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[3214]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[3225]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[3233]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[3264]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[3315]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[3327]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[3334]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[3419]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[3447]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[3448]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[3453]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[3464]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[3475]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[3589]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[3607]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[3609]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[3751]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[3758]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[3761]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[3841]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[3855]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[3955]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[3958]: Null check (P13: 9.90 1/3)
D:\target\0001\fast\Configuration.java[225]: Non final class (P24: 2.00 2/3)
D:\target\0001\fast\Configuration.java[244]: Non final class (P24: 2.00 2/3)
D:\target\0001\fast\Configuration.java[368]: Non final class (P24: 2.00 2/3)
D:\target\0001\fast\Configuration.java[423]: Non final class (P24: 2.00 2/3)
D:\target\0001\fast\Configuration.java[463]: Non final class (P24: 2.00 2/3)
D:\target\0001\fast\Configuration.java[2069]: Non final class (P24: 2.00 2/3)
D:\target\0001\fast\Configuration.java[2070]: Non final class (P24: 2.00 2/3)
D:\target\0001\fast\Configuration.java[2075]: Non final class (P24: 2.00 2/3)
D:\target\0001\fast\Configuration.java[3086]: Non final class (P24: 2.00 2/3)
D:\target\0001\fast\Configuration.java[3106]: Non final class (P24: 2.00 2/3)
D:\target\0001\fast\Configuration.java[1280]: Count If Return (P6: 1.30 3/3)
D:\target\0001\fast\Configuration.java[1842]: Count If Return (P6: 1.30 3/3)
D:\target\0001\fast\Configuration.java[1852]: Count If Return (P6: 1.30 3/3)
D:\target\0001\fast\Configuration.java[2050]: Count If Return (P6: 1.30 3/3)
D:\target\0001\fast\Configuration.java[2054]: Count If Return (P6: 1.30 3/3)
D:\target\0001\fast\Configuration.java[2090]: Count If Return (P6: 1.30 3/3)
D:\target\0001\fast\Configuration.java[2100]: Count If Return (P6: 1.30 3/3)
D:\target\0001\fast\Configuration.java[2272]: Count If Return (P6: 1.30 3/3)
D:\target\0001\fast\Configuration.java[2322]: Count If Return (P6: 1.30 3/3)
D:\target\0001\fast\Configuration.java[2601]: Count If Return (P6: 1.30 3/3)
D:\target\0001\fast\Configuration.java[2612]: Count If Return (P6: 1.30 3/3)
D:\target\0001\fast\Configuration.java[2822]: Count If Return (P6: 1.30 3/3)
D:\target\0001\fast\Configuration.java[2846]: Count If Return (P6: 1.30 3/3)
D:\target\0001\fast\ErrorExample.java: error when calculating patterns: Can't count P1 metric: 
D:\target\0001\fast\JndiRmiClientInterceptor.java score: 42.80000000000001
D:\target\0001\fast\JndiRmiClientInterceptor.java[271]: Multiple try (P11: 17.80 1/11)
D:\target\0001\fast\JndiRmiClientInterceptor.java[365]: Multiple try (P11: 17.80 1/11)
D:\target\0001\fast\JndiRmiClientInterceptor.java[197]: Instance of (P8: 7.60 2/11)
D:\target\0001\fast\JndiRmiClientInterceptor.java[201]: Instance of (P8: 7.60 2/11)
D:\target\0001\fast\JndiRmiClientInterceptor.java[366]: Instance of (P8: 7.60 2/11)
D:\target\0001\fast\JndiRmiClientInterceptor.java[389]: Instance of (P8: 7.60 2/11)
D:\target\0001\fast\JndiRmiClientInterceptor.java[198]: String concat (P17: 5.60 3/11)
D:\target\0001\fast\JndiRmiClientInterceptor.java[202]: String concat (P17: 5.60 3/11)
D:\target\0001\fast\JndiRmiClientInterceptor.java[229]: String concat (P17: 5.60 3/11)
D:\target\0001\fast\JndiRmiClientInterceptor.java[277]: String concat (P17: 5.60 3/11)
D:\target\0001\fast\JndiRmiClientInterceptor.java[323]: String concat (P17: 5.60 3/11)
D:\target\0001\fast\JndiRmiClientInterceptor.java[326]: String concat (P17: 5.60 3/11)
D:\target\0001\fast\JndiRmiClientInterceptor.java[378]: String concat (P17: 5.60 3/11)
D:\target\0001\fast\JndiRmiClientInterceptor.java[415]: String concat (P17: 5.60 3/11)
D:\target\0001\fast\JndiRmiClientInterceptor.java[120]: Setters (P2: 5.00 4/11)
D:\target\0001\fast\JndiRmiClientInterceptor.java[137]: Setters (P2: 5.00 4/11)
D:\target\0001\fast\JndiRmiClientInterceptor.java[148]: Setters (P2: 5.00 4/11)
D:\target\0001\fast\JndiRmiClientInterceptor.java[163]: Setters (P2: 5.00 4/11)
D:\target\0001\fast\JndiRmiClientInterceptor.java[175]: Setters (P2: 5.00 4/11)
D:\target\0001\fast\JndiRmiClientInterceptor.java[200]: Null check (P13: 2.70 5/11)
D:\target\0001\fast\JndiRmiClientInterceptor.java[246]: Null check (P13: 2.70 5/11)
D:\target\0001\fast\JndiRmiClientInterceptor.java[250]: Null check (P13: 2.70 5/11)
D:\target\0001\fast\JndiRmiClientInterceptor.java[369]: Force Type Casting (P5: 2.40 6/11)
D:\target\0001\fast\JndiRmiClientInterceptor.java[390]: Force Type Casting (P5: 2.40 6/11)
D:\target\0001\fast\JndiRmiClientInterceptor.java[245]: Count If Return (P6: 0.60 7/11)
D:\target\0001\fast\JndiRmiClientInterceptor.java[288]: Count If Return (P6: 0.60 7/11)
D:\target\0001\fast\JndiRmiClientInterceptor.java[321]: Count If Return (P6: 0.60 7/11)
D:\target\0001\fast\JndiRmiClientInterceptor.java[76]: Implements Multi (P7: 0.60 8/11)
D:\target\0001\fast\JndiRmiClientInterceptor.java[246]: Return null (P16: 0.20 9/11)
D:\target\0001\fast\JndiRmiClientInterceptor.java[292]: Empty Rethrow (P3: 0.20 10/11)
D:\target\0001\fast\JndiRmiClientInterceptor.java[76]: Prohibited class name (P4: 0.10 11/11)
D:\target\0001\fast\LottieImageAsset.java score: 0.6
D:\target\0001\fast\LottieImageAsset.java[17]: Non final attribute (P12: 0.60 1/1)
Total average score: 18.86666666666667

Process finished with exit code 1

```

File: aibolit/ml_pipeline/ml_pipeline.py [hunk 1]
```diff
@@ -116,7 +117,8 @@ def train_process():
         model_new = pickle.load(fid)
         print('Model has been loaded successfully')
         for x in X_test:
-            preds, importances = model_new.predict(x)
+            snippet = [i for i in x] + [randint(1, 200)]
```
Comment (acheshkov) [inline aibolit/ml_pipeline/ml_pipeline.py:120]: What randint(100) do?

Comment (lyriccoder) [reply aibolit/ml_pipeline/ml_pipeline.py:120]: it's just a random number. New interface demands ncss at the end of snippter. But training dataset doesn't have ncss. We have only patterns. And that part of code relates to smoke testing after training a dataset

**Comment by acheshkov** [comment]
@rultor merge

**Comment by 0crat** [comment]
Job `gh:yegor256/aibolit#444` is not assigned, can't get performer

**Comment by acheshkov** [comment]
@rultor release, tag=`1.1.2rc1`

**Comment by acheshkov** [comment]
@rultor release, tag=`1.1.2`

**Pull request merged**